common/src/main/java/org/valkyrienskies/mod/mixin/entity/MixinEntity.java

```diff
@@ -211,28 +210,6 @@ private void preCalculateViewVector(final float xRot, final float yRot, final Ca
     }
     */
 
-    /**
-     * @reason Without this and that other mixin, things don't render correctly at high speeds.
-     * @see org.valkyrienskies.mod.mixin.client.renderer.MixinEntityRenderer
-     */
-    @Inject(method = "shouldRender", at = @At("HEAD"), cancellable = true)
-    private void onShouldRender(double d, double e, double f, CallbackInfoReturnable<Boolean> cir) {
-        if (this.draggingInformation.isEntityBeingDraggedByAShip() && this.level.isClientSide) {
-            final ClientShip ship = VSGameUtilsKt.getShipObjectWorld((ClientLevel) this.level).getAllShips().getById(this.draggingInformation.getLastShipStoodOn());
```

Comment:
I believe you should do `VSGameUtilsKt.getShipObjectWorld(this.level).getShip... instanceof ClientShip ship` instead of create a separate mixin or use force casting. One should not assume `this.level` is always `ClientLevel` for compatible reasons.

Reply:
importing ClientLevel crashes the server